### ISSUE TITLE
Add `macos` & `linux` modules

### ISSUE THREAD

**Issue by Fizzadar** [Fizzadar/pyinfra#960]
Move `server.user` -> `linux.user` and add `macos.user` (using `sysadminctl`). Then have `server.user` behave like `server.packages` and similar where we select `[linux|macos|windows].user` according to the remote OS.

**Comment by v3ss0n** [comment]
makes sense

**Comment by gnat** [comment]
Good idea. Wouldn't want this to turn into a rabbithole, but would be nice to have a variable somewhere that tells you which platform you're on, so you can branch inside the scripts.

Easy way for linux: `uname`
